src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -412,6 +414,81 @@ def exponential_backoff_with_jitter(
     return random.uniform(0, exponential_backoff)
 
 
+class Session(SqlModelSession):
+    """Session subclass that automatically tracks duration and calling context."""
+
+    def __enter__(self) -> "Session":
+        """Enter the context manager.
+
+        Returns:
+            The SqlModel session.
+        """
+        if logger.isEnabledFor(logging.DEBUG):
```

Comment:
Should we maybe wrap this whole thing in a `try...except` to make sure it never fails? Or do we want to catch failures in here anyway?

Reply:
Ah, good catch !

Reply:
On second thought, I see no reason why exceptions here should be expected or ignored for that matter.